R/dictionary.R

```diff
@@ -44,15 +44,16 @@ vec_count <- function(x, sort = c("count", "key", "location", "none")) {
   # Returns key-value pair giving index of first occurence value and count
   kv <- .Call(vctrs_count, vec_proxy_equal(x))
 
-  df <- data.frame(key = 0, count = kv$val)
+  # rep_along() to support zero-length vectors!
+  df <- data.frame(key = rep_along(kv$val, NA), count = kv$val)
   df$key <- vec_slice(x, kv$key) # might be a dataframe
 
   if (sort == "none")
     return(df)
 
   idx <- switch(sort,
     location = order(kv$key),
-    key = order(df$key),
+    key = order_proxy(vec_proxy_compare(df$key)),
```

Comment:
Why did this change?

Reply:
> vec_count() now works in corner cases and for data frames with sort = "key"

``` r
withr::with_options(list(max.print = 100), print(vctrs::vec_count(
  iris,
  "key"
)))
#>    key.Sepal.Length key.Sepal.Width key.Petal.Length key.Petal.Width
#> 1                NA              NA               NA              NA
#> 2                NA              NA               NA              NA
#> 3                NA              NA               NA              NA
#> 4                NA              NA               NA              NA
#> 5                NA              NA               NA              NA
#> 6                NA              NA               NA              NA
#> 7                NA              NA               NA              NA
#> 8                NA              NA               NA              NA
#> 9                NA              NA               NA              NA
#> 10               NA              NA               NA              NA
#> 11               NA              NA               NA              NA
#> 12               NA              NA               NA              NA
#> 13               NA              NA               NA              NA
#> 14               NA              NA               NA              NA
#> 15               NA              NA               NA              NA
#> 16               NA              NA               NA              NA
#>    key.Species count
#> 1         <NA>    NA
#> 2         <NA>    NA
#> 3         <NA>    NA
#> 4         <NA>    NA
#> 5         <NA>    NA
#> 6         <NA>    NA
#> 7         <NA>    NA
#> 8         <NA>    NA
#> 9         <NA>    NA
#> 10        <NA>    NA
#> 11        <NA>    NA
#> 12        <NA>    NA
#> 13        <NA>    NA
#> 14        <NA>    NA
#> 15        <NA>    NA
#> 16        <NA>    NA
#>  [ reached getOption("max.print") -- omitted 34 rows ]
#>  [ reached 'max' / getOption("max.print") -- omitted 695 rows ]
```

<sup>Created on 2019-01-25 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>